staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -127,13 +127,17 @@ func NewReflector(lw ListerWatcher, expectedType interface{}, store Store, resyn
 
 // NewNamedReflector same as NewReflector, but with a specified name for logging
 func NewNamedReflector(name string, lw ListerWatcher, expectedType interface{}, store Store, resyncPeriod time.Duration) *Reflector {
+	realClock := &clock.RealClock{}
 	r := &Reflector{
 		name:          name,
 		listerWatcher: lw,
 		store:         store,
-		period:        time.Second,
-		resyncPeriod:  resyncPeriod,
-		clock:         &clock.RealClock{},
+		// We used to make the call every 1sec (1 QPS), the goal here is to achieve ~98% traffic reduction when
+		// API server is not healthy. With these parameters, backoff will stop at [30,60) sec interval which is
+		// 0.22 QPS.
+		backoffManager: wait.NewExponentialBackoffManager(100*time.Millisecond, 30*time.Second, 2*time.Minute, 2.0, 1.0, realClock),
```

Comment:
hm, that 100ms initial (vs 1s) might make the first couple seconds worse than before-- what do you think about putting that at 800ms?

Reply:
800ms can help in the api server crash case but would make reconnecting not so agile during normal connection broken (which in production is the most of the re-connect we are seeing at Pinterest)

100ms is more likely to screw up in api server crash case (i.e. oom killed, when it gets restarted, all api calls can squeeze in), but since there are exponential backoff, what it does is to (sort of) delay the recovery of api server by 10s seconds to minutes.

Reply:
inputs @wojtek-t @liggitt ?

Reply:
If your clients are reconnecting often enough that this 1s is a problem, something else is wrong, you have a watch cache configuration problem or etcd size problem or something.

I think 800ms is a safe improvement over 1s (that is, it's mostly compensated by the jitter you added).

Reply:
sounds good. Lets' go with 800ms. I updated the PR